internal/controller/isbservicerollout_controller.go

```diff
@@ -215,13 +196,17 @@ func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceR
 	return nil
 }
 
-func (r *ISBServiceRolloutReconciler) calculateChildSpecHash(ctx context.Context, obj *kubernetes.GenericObject) (string, error) {
+func (r *ISBServiceRolloutReconciler) shouldUpdateCR(ctx context.Context, obj *kubernetes.GenericObject) (bool, error) {
```

Comment:
why not pass in the object directly and only check this when a CR exist?

Reply:
Good point. I'll actually move the entire comparison logic inside ApplyCRSpec since we are only comparing specs directly now.